src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -386,20 +386,19 @@ private void initMappers(Map<String, Object> withoutType) {
         } else {
             this.timestamp = Timestamp.EMPTY;
         }
+
         if (withoutType.containsKey("_parent")) {
-            this.hasParentField = true;
-        } else {
-            this.hasParentField = false;
+            this.parentType = ((Map<String, String>) withoutType.get("_parent")).get("type");
```

Comment:
I don't think it's legal to cast to a Map&lt;String, String&gt; directly since the parent mapping can also take a map of fielddata settings? So we should first case the map to Map&lt;String, String&gt; and then case the value from Object to String?